updatem/domain/update_manager_test.go

```diff
@@ -0,0 +1,409 @@
+// Copyright (c) 2023 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+package domain
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/eclipse-kanto/update-manager/api"
+	"github.com/eclipse-kanto/update-manager/api/types"
+	"github.com/eclipse-kanto/update-manager/test/mocks"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	testDomain         = "test-domain"
+	testActivityID     = "testActivityId"
+	initCurrentStateID = "initial-current-state"
+)
+
+var dummyInventory *types.Inventory = &types.Inventory{
+	HardwareNodes: []*types.HardwareNode{
+		{
+			InventoryNode: types.InventoryNode{
+				ID:      "hardware-node-id",
+				Version: "1.0.0",
+				Name:    "Hardware Node",
+				Parameters: []*types.KeyValuePair{
+					{
+						Key:   "x",
+						Value: "y",
+					},
+				},
+			},
+			Addressable: true,
+		},
+	},
+	SoftwareNodes: []*types.SoftwareNode{
+		{
+			InventoryNode: types.InventoryNode{
+				ID:      "software-node-id",
+				Version: "1.0.0",
+				Name:    "Software Node",
+				Parameters: []*types.KeyValuePair{
+					{
+						Key:   "x",
+						Value: "y",
+					},
+				},
+			},
+			Type: types.SoftwareTypeApplication,
+		},
+	},
+	Associations: []*types.Association{
+		{
+			SourceID: "hardware-node-id",
+			TargetID: "software-node-id",
+		},
+	},
+}
+
+var domain *types.Domain = &types.Domain{
+	ID: testDomain,
+}
+
+var desiredState *types.DesiredState = &types.DesiredState{
+	Domains: []*types.Domain{domain},
+}
+
+var command *types.DesiredStateCommand = &types.DesiredStateCommand{
+	Baseline: "dummyBaseline",
+	Command:  "dummyCommand",
+}
+
+func TestUpdateManagerName(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	desiredStateClient := mocks.NewMockDesiredStateClient(mockCtrl)
+	desiredStateClient.EXPECT().Domain().Return(testDomain)
+
+	updateManager := createTestDomainUpdateManager(desiredStateClient, nil)
+	assert.Equal(t, testDomain, updateManager.Name())
+}
+
+func TestGetCurrentState(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	eventCallback := mocks.NewMockUpdateManagerCallback(mockCtrl)
+	eventCallback.EXPECT().HandleCurrentStateEvent(testDomain, initCurrentStateID, dummyInventory)
+
+	updateManager := createTestDomainUpdateManager(nil, eventCallback)
+	updateManager.readTimeout = time.Second
+	updateManager.currentState = &internalCurrentState{}
```

Comment:
Those three lines are repeated all over the test. `createTestDomainUpdateManager(...)` should be extended with more parameters to avoid unnecessary repetitions